src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -576,13 +603,43 @@ export function findPreferredKernel(
                         metadata.interpreter?.path?.toLowerCase().includes('python3') ||
                         spec.argv[0].toLocaleLowerCase().includes('python3'))
                 ) {
+                    traceInfoIf(isCI, 'Increased score by +1 for matching major version 3');
                     score += 1;
                     subScore += 1;
                     traceInfo(
                         `findPreferredKernel score for Python3, ${getDisplayNameOrNameOfKernelConnection(
                             metadata
                         )} is ${score}`
                     );
+
+                    // If the versions match exactly, then increase the score by another 1
+                    if (
+                        typeof notebookMetadata === 'object' &&
+                        'interpreter' in notebookMetadata &&
+                        notebookMetadata.interpreter &&
+                        typeof notebookMetadata.interpreter === 'object' &&
+                        'hash' in notebookMetadata.interpreter &&
+                        metadata.kind === 'startUsingPythonInterpreter' &&
+                        notebookMetadata.interpreter
```

Comment:
notebookMetadata.interpreter looks repeated here. Also is the 'hash' needed?

Reply:
Thanks